### ISSUE TITLE
App Engine asset deployment (dos, cron etc.) should be available for flex as well

### ISSUE THREAD

**Issue by etanshaul** [GoogleCloudPlatform/cloud-code-intellij#1373]
we enabled it for standard, but we should allow it for flex as well

**Comment by patflynn** [comment]
Before we can do this properly we'll have to give the user the option of
specifying the App Engine configuration directory rather than just a path
to the app.yaml

**Comment by etanshaul** [comment]
ah yes. good point. my PR for this was a bit premature.

**Comment by etanshaul** [comment]
is there a "standard" place for putting these that we can look as a first step? or is it completely up to the user.

**Comment by patflynn** [comment]
by default maven and gradle look in src/main/appengine, but it's also
configurable.

**Comment by etanshaul** [comment]
ok cool. this needs a bit more thought then. like where we specify this (is it a path selector next to the asset checkbox, does it belong in the facet, etc.)

**Comment by etanshaul** [comment]
@patflynn 

My thinking is that belongs in the flex facet, together with the app.yaml and Dockerfile configuration. This will require additional validation in the deploy config to ensure that the path was specified in the facet when the user chooses to deploy these configs (or it could just attempt to find them in src/main/appengine directory if none was specified).

This config deploy feature, however, was added to 17.2 before the flex facet was added. So unless you can think of another strategy, I think we'll need to wait until we release the flex updates for this change to get released. In the meantime, the users will be able to specify these configs for standard only.

**Comment by patflynn** [comment]
That's OK. We can hold off on adding this to flex for now.

**Comment by etanshaul** [comment]
@patflynn I'm finishing up this issue and was wondering:

For standard, the additional assets get staged into `WEB-INF/appengine-generated/` at the root of the staging directory. We then look for them here if the user selects to deploy all configs.

I'm wondering if for flex, we should also stage these files. As I see it there are 2 options:

1) We don't stage the files, and if the user selected to deploy all configs then, at deploy time, we find these files in the app engine config folder (see #1497)
2) At the flex stage step, we copy the files from the appengine config folder into the staging directory (either at the root together with the app.yaml and the artifact, or in some subdirectory) and then pull these assets from here at deploy time.

Wanted to get your opinion on these. And if 2 makes more sense, is there a preferred place to put these assets in the staging directory?

My thought on this is that it doesn't really matter and that option 1 is fine (since the staging directory is temporary), but wanted your opinion in case I'm overlooking something.

**Comment by patflynn** [comment]
Don't stage the files. It's not meaningful/necessary in the same way the
app.yaml relates to the project artifacts in the deployment directory, and
it creates a risk that the user will accidentally include the files in
their serving path.

**Comment by etanshaul** [comment]
Cool, thanks for confirming.

**Comment by etanshaul** [comment]
closed as "won't fix" due to limited enhancements for ae flex
